sdks/python/apache_beam/transforms/core.py

```diff
@@ -189,6 +273,63 @@ def finish_bundle(self):
     """
     pass
 
+  def new_tracker(self, restriction):
+    """Produces a new ``RestrictionTracker`` for the given restriction.
+
+    Args:
+      restriction: an object that defines a restriction as identified by the
+        current ``DoFn``. For example, a tuple that gives a range of positions for
+        a ``DoFn`` that reads files based on byte positions.
+    Returns: an object of type ``RestrictionTracker``.
+    """
+    pass
+
+  def restriction_coder(self):
+    """Returns a ``Coder`` for restrictions produced by the current ``DoFn``.
+
+    Returns:
+      an object of type ``Coder``.
+    """
+    pass
+
+  def initial_restriction(self, element):
+    """Produces an initial restriction for the given element."""
+    pass
+
+  def split(self, element, restriction):
+    """Splits the given element and restriction.
+
+    Returns an iterator of restrictions. The total set of elements produced by
+    reading input element for each of the returned restrictions should be the
+    same as the total set of elements produced by reading the input element for
+    the input restriction.
+
+    TODO(chamikara): give suitable hints for performing splitting, for example
+    number of parts or size in bytes.
+    """
+    pass
+
+  @staticmethod
+  def stop():
```

Comment:
Is this necessary in Python? Seems like with the "last yielded element is a ProcessContinuation" thing, instead of saying yield stop() you can simply not yield anything.

Reply:
Yeah, makes sense to remove this. Removed and updated docs.